src/executer.ts

```diff
@@ -0,0 +1,57 @@
+import { TaskQueue } from './types'
+
+const QUEUE = 'commandQueue'
+const AGENTS = 'agents'
+
+export default class Executer {
+  private createAgent: Function
+  private commandQueue: any[]
+  private agents: any
+
+  constructor (Agent: any) {
+    this.createAgent = () => new Agent()
+    this.commandQueue = []
+    this.agents = {}
+  }
+
+  run (args: any) {
+    let cmd: TaskQueue[] = this.parse(args)
+    cmd = this.commandQueue.concat(cmd)
```

Comment:
`this[QUEUE] = [this[QUEUE], ...cmd]`

https://qiita.com/mizchi/items/4197a8d945223545e27a#array